src/assets/Standards/CollectibleStandards/ERC721/ERC721Standard.test.ts

```diff
@@ -509,18 +459,45 @@ describe('ERC721Standard', () => {
       })
       .reply(200, {
         jsonrpc: '2.0',
-        id: 21,
+        id: 20,
         result:
           '0x0000000000000000000000000000000000000000000000000000000000000000',
+      });
+
+    const result = async () => {
+      await erc721Standard.getDetails(CRYPTO_KITTIES_ADDRESS, '4');
+    };
+    await expect(result).rejects.toThrow("This isn't a valid ERC721 contract");
```

Comment:
Does this mean that transfering CryptoKitties will fail in the same vain as we've seen with this issue elsewhere? If so, that seems bad.

Reply:
Agreed. Displaying a contract interaction with a method that _happens_ to be called "approve" but does something different than an ERC20 approve could be dangerous and misleading.

If a contract doesn't match a standard, I'm wary of us treating it like it does.

Reply:
I agree that `ERC721Standard` doesn't need to solve all the problems. But if it has known limitations for popular tokens we should come up with a solution to allow users to interact with these popular tokens. Furthermore, we should **definitely** never show an infinite loading just because this controller logic fails.

Reply:
But just to clarify @Gudahtt, whether this code returns that the contract fits a standard is no guarantee that the contract is not malicious. There is no guarantee to what `approve()` does regardless.

Reply:
> we should come up with a solution to allow users to interact with these popular tokens

Do you have thoughts about the proposal above 

> to create a "non-standard-class" that we can fall back to and return any common basic properties (symbol, name, decimals, tokenURI, image) that crop up across non-standard versions of the most common contract standards (ERC20, ERC721, ERC1155)

If the contract is not one of the 3 major standards we try to extract `name`, `symbol`, `decimal`, `tokenURI` and whatever else we think is common enough and just return with a field `standard: unknown`? 

@Gudahtt @FrederikBolding 

regarding:
> we should definitely never show an infinite loading just because this controller logic fails

I think we need some design or at least copy for a screen for the situation where someone calls `approve` and the controllers failed to retrieve `name` or `symbol`... Does that make sense?

Reply:
>whether this code returns that the contract fits a standard is no guarantee that the contract is not malicious. There is no guarantee to what approve() does regardless.

:100: 

My comment was phrased poorly. I meant even if we assume the contract is not malicious, it could be dangerous and misleading to assume a method called `approve` is equivalent to an ERC20 approval. For a malicious contract, we can trust nothing, as you said.

Reply:
>I think we need some design or at least copy for a screen for the situation where someone calls approve and the controllers failed to retrieve name or symbol... Does that make sense?

We had this already though, didn't we? I thought this was a recent regression. I was guessing it was related in some way to the switch from the preferences controller to the shared token controller, or some other related change.

Reply:
regarding your comment here @Gudahtt :

> Displaying a contract interaction with a method that happens to be called "approve" but does something different than an ERC20 approve could be dangerous and misleading

We're actually not making any guarantees about the implementation, just that the contract interface matches a certain standard. So, to be honest, in insisting on adherence to standards we're not really protecting against misleading the user about what the action will do (that risk always potentially exists), we're simply organizing our code around the existence and prevalence of these standards

Reply:
> We had this already though, didn't we? I thought this was a recent regression.
🤔  Possibly. I wasn't aware of it if it did exists but I can look into it!

Reply:
Interestingly though, we are failing on CryptoKitties because they don't implement the entire ERC721 standard (not sure what the history is there because they call it 721 in the Solidity). But we don't use any of the ERC721 standard methods after wards. CryptoKitties has both a valid `name()` and `symbol()` 🤔

Reply:
I mean yes we don't currently use the methods that are part of the ERC721 standard which the CryptoKitties contract fails to implement (`setApprovalForAll`, `getApproved`, `ApprovalForAll`, `isApprovedForAll`) but we may eventually want to make use of them and we would be setting ourselves up for confusion if we hand wave it away that the contract almost fully implements the contract?

Reply:
I guess so - That shouldn't be a problem when doing an `approve()` though. This may be something that should be solved elsewhere as this PR is already an improvement to what we had before!

Reply:
And to respond to your previous question @adonesky1 - I think it would be interesting to try your "non-standard-class" method as long as we add something in the UI to tell the user we are not completely sure about the contract standard. But trying to gather some data using "semi-standard" function calls would be an interesting way to do it 👍